### PR TITLE
Features/experimental brushes export

### PR DESCRIPTION
**Pull request by andybak**
So - I'm not sure how to progress this. I didn't fully understand all the changes in the PR when I made them. It was a quick attempt to get experimental brushes exporting in some form. My hunch is that it's an improvement on what we've got and it's no use keeping it hidden on a branch. Tricky one to test but hard to see how it could be worse than the status quo.

### CODE REVIEW

File: Assets/Resources/X/Brushes/DryBrush/DryBrush.asset [hunk 1]
```diff
@@ -14,7 +14,7 @@ MonoBehaviour:
   m_EditorClassIdentifier: 
   m_Guid:
     m_storage: 725f4c6a-6427-6524-29ab-da371924adab
-  m_DurableName: 
+  m_DurableName: DryBrush
```
Comment (mikeage) [inline Assets/Resources/X/Brushes/DryBrush/DryBrush.asset:17]: We have a few brushes that still don't have `m_DurableName` set. Is this deliberate?

```
Assets/Resources/Brushes/Basic/Felt/Felt.asset:  m_DurableName:
Assets/Resources/Brushes/Basic/TaperedHighlighter/TaperedHighlighter.asset:  m_DurableName:
Assets/Resources/Brushes/New BrushDescriptor.asset:  m_DurableName:
Assets/Resources/X/Brushes/Test/TubeDiffuse.asset:  m_DurableName:
Assets/Resources/X/Brushes/Test/TubeUnlit.asset:  m_DurableName:
```

Comment (andybak) [reply Assets/Resources/X/Brushes/DryBrush/DryBrush.asset:17]: Not sure. I think my strategy was "export > fix > repeat" and it's possible the test scene I had wasn't using all brushes. I wonder if those brushes aren't user-accessible? That would be the reason they would be missing from the test scene.

Comment (andybak) [reply Assets/Resources/X/Brushes/DryBrush/DryBrush.asset:17]: I'll fix 'em anyway as they can be accessed via the API

Comment (mikeage) [reply Assets/Resources/X/Brushes/DryBrush/DryBrush.asset:17]: gotcha!

**Comment by andybak** [comment]
Making this non-draft in a passive-aggressive attempt to get some feedback. ;-)

**Comment by mikeage** [comment]
Want to announce this on discord and see if a few people who use experimental mode want to give it a shot?

From what I understand, it looks reasonable; as you say, no worse than current.

File: Assets/Scripts/Export/ExportFbx.cs [hunk 2]
```diff
@@ -194,7 +194,7 @@ class ExportMesh {
       try {
         WriteObjectsAndConnections2(G);
         G.m_exporter.Export(G.m_scene);
-      } catch (InvalidOperationException e) {
+      } catch (GlbError e) {
```
Comment (merulaxr) [inline Assets/Scripts/Export/ExportFbx.cs:197]: GlbError for FBX export?

Comment (andybak) [reply Assets/Scripts/Export/ExportFbx.cs:197]: Yeah. One of the problems is that I did this way back as a quick experiment to see if I could get export working. 

I suspect that might be my past self going "let's put an exception in here that won't happen as it's quicker than commenting out the whole block". Exactly the kind of sloppy code I was hoping this review would catch ;-)

I'll remove that change and see what I was actually aiming to achieve here.

Have you taken a look at my shader edits? There's probably some bad things lurking there as well.

**Review by billyquith — APPROVED** [review]
Untested by me. No reason to not approve.

File: Assets/Scripts/Export/ExportFbx.cs (hunk 2 above)
Comment (andybak) [reply Assets/Scripts/Export/ExportFbx.cs]: Fixed

**Review by merulaxr — APPROVED** [review]
Does export, though I'm not sure what you'd use it with at this point! Someone can help me convert these to gles 3 shaders :)

**Pull request merged**